### PR TITLE
Created an announcement of Ember 3.22 release

### PR DESCRIPTION
**Pull request by ijlee2**

### CODE REVIEW

**Comment by locks** [comment]
## emberjs

- [x] [#19062](https://github.com/emberjs/ember.js/pull/19062) / [#19068](https://github.com/emberjs/ember.js/pull/19068) [FEATURE] Add @ember/destroyable feature from the [Destroyables RFC](https://github.com/emberjs/rfcs/blob/master/text/0580-destroyables.md).
  - link to API docs/RFC
  - don't go into much detail
- [x] [#18984](https://github.com/emberjs/ember.js/pull/18984) / [#19067](https://github.com/emberjs/ember.js/pull/19067) [FEATURE] Add low-level Cache API per [Autotracking Memoization RFC](https://github.com/emberjs/rfcs/blob/master/text/0615-autotracking-memoization.md)
  - already mentioned in [the Guides](https://guides.emberjs.com/release/in-depth-topics/autotracking-in-depth/#toc_caching-of-tracked-properties)
  - https://github.com/ember-learn/guides-source/pull/1474#issuecomment-668029558
  - polyfill for <3.22. check if Guides mention it
  - available by default in 3.22 (ed: feature flag turned on)
- [x] [#19086](https://github.com/emberjs/ember.js/pull/19086) [FEATURE] Pass transition object to activate/deactivate hooks and events (🔏 @ijlee2)
  - you can now use the transition object in activate/deactive Route hooks
- [x] [#19094](https://github.com/emberjs/ember.js/pull/19094) [BUGFIX] Fix RouterService#isActive() to work with tracking (🔏  @jaredgalanis)
  - RouterService#isActive now updates tracked properties correctly 
- [x] [#19170](https://github.com/emberjs/ember.js/pull/19170) [BUGFIX] Make modifier manager 3.22 accept the resolved value directly.
   - modifier manager 3.22 introduced ("addon developers")
- [x] [#19124](https://github.com/emberjs/ember.js/pull/19124) [BUGFIX] Fix rendering engine usage within a `fastboot` sandbox
  - fixes fastboot support for 3.22

## ember data

- [x] [#7126](https://github.com/emberjs/data/pull/7126) Bugfix for wrong hasMany relationship state in scenario of sideposting with lid
- [x] [#7273](https://github.com/emberjs/data/pull/7273) [BUGFIX] Entangle Errors.errorsFor properly
- [x] [#7322](https://github.com/emberjs/data/pull/7322) [Build]: disable esm cache (#7322)
  - if you had errors during build this might fix it
- [x] [#7330](https://github.com/emberjs/data/pull/7330) [BUG]: Consume array access to autotrack hasMany
- [x] [#7335](https://github.com/emberjs/data/pull/7335) [Bug]: errors `remove` to accurately update in template
  - `errorsFor` did not properly update in templates.
  - pzuraq/snewcomer for context

## ember-cli

- [ ] Possibly mention eslint-plugin-ember 9.0.0 breaking changes?

File: source/2020-10-06-ember-3-22-released.md
Comment (ijlee2) [inline source/2020-10-06-ember-3-22-released.md]:
```suggestion
<!-- TODO: List features -->
- Placeholder for `@ember/destroyable` RFC
- Placeholder for Cache API
```

File: source/2020-10-06-ember-3-22-released.md
Comment (ijlee2) [inline source/2020-10-06-ember-3-22-released.md]:
```suggestion
- Enabled `@ember/destroyable` API, which allows users to create and manage their own destroyables. For more information, please see the [Destroyables RFC](https://emberjs.github.io/rfcs/0580-destroyables.html).
```

File: source/2020-10-06-ember-3-22-released.md
Comment (ijlee2) [inline source/2020-10-06-ember-3-22-released.md]:
```suggestion
- Enabled Cache API, a low-level primitive for memoizing the result of a function based on autotracking. (You can install a [polyfill](https://github.com/ember-polyfills/ember-cache-primitive-polyfill) for apps between versions 3.13 and 3.21.) For more information, please check the [Ember Guides](https://guides.emberjs.com/release/in-depth-topics/autotracking-in-depth/#toc_caching-of-tracked-properties) and [Autotracking Memoization RFC](https://emberjs.github.io/rfcs/0615-autotracking-memoization.html).
```

File: source/2020-10-06-ember-3-22-released.md
Comment (ijlee2) [inline source/2020-10-06-ember-3-22-released.md]:
```suggestion
- Enabled `@ember/destroyable` API, which allows you to create and manage your own destroyables. For more information, please see the [Destroyables RFC](https://emberjs.github.io/rfcs/0580-destroyables.html).
```

File: source/2020-10-06-ember-3-22-released.md
Comment (ijlee2) [inline source/2020-10-06-ember-3-22-released.md]:
```suggestion
author: Jared Galanis, Isaac Lee, Ricardo Mendes
```

File: source/2020-10-06-ember-3-22-released.md
Comment (snewcomer) [inline source/2020-10-06-ember-3-22-released.md]:
```suggestion
- [a fix](https://github.com/emberjs/data/pull/7126) that addresses relationship state when sideposting with `lid`.
- [a fix](https://github.com/emberjs/data/pull/7330) that ensures changes to hasMany relationships are autotracked.
```

File: source/2020-10-06-ember-3-22-released.md
Comment (ijlee2) [inline source/2020-10-06-ember-3-22-released.md]:
```suggestion
Ember.js 3.22 introduced 3 features.

<!-- TODO: List features -->
- Enabled `@ember/destroyable` API, which allows you to create and manage your own destroyables. For more information, please see the [Destroyables RFC](https://emberjs.github.io/rfcs/0580-destroyables.html).
<!-- alex disable retext-equality -->
- Enabled Cache API, a low-level primitive for memoizing the result of a function based on autotracking. (You can install a [polyfill](https://github.com/ember-polyfills/ember-cache-primitive-polyfill) for apps between versions 3.13 and 3.21.) For more information, please check the [Ember Guides](https://guides.emberjs.com/release/in-depth-topics/autotracking-in-depth/#toc_caching-of-tracked-properties) and [Autotracking Memoization RFC](https://emberjs.github.io/rfcs/0615-autotracking-memoization.html).
- Allowed passing the `transition` object to 2 route hooks, `activate` and `deactivate`.
```

File: source/2020-10-19-ember-3-22-released.md
Comment (ijlee2) [inline source/2020-10-19-ember-3-22-released.md]:
```suggestion
Ember CLI 3.22 includes only internal changes. However, please note that [eslint-plugin-ember](https://github.com/ember-cli/eslint-plugin-ember) released a new major version, `v9.x`. The new version introduces 6 additional recommended rules and removed support of Node `v13`. To learn more, please [check out the changelog for `eslint-plugin-ember`](https://github.com/ember-cli/eslint-plugin-ember/releases/tag/v9.0.0).


#### New Features

Ember CLI 3.22 introduced 0 features.

#### Deprecations

Ember CLI 3.22 introduced 0 deprecations.
```

File: source/2020-10-19-ember-3-22-released.md
Comment (ijlee2) [inline source/2020-10-19-ember-3-22-released.md]:
```suggestion
Ember CLI 3.22 includes only internal changes. However, please note that [eslint-plugin-ember](https://github.com/ember-cli/eslint-plugin-ember) released a new major version, `v9.x`. The new version introduces 6 recommended rules and removes support of Node `v13`.

New Ember apps will install `v9` or higher of `eslint-plugin-ember`, which may limit the version of Node that you use. To learn more, please [check out the changelog for `eslint-plugin-ember`](https://github.com/ember-cli/eslint-plugin-ember/releases/tag/v9.0.0).
```

File: source/2020-10-19-ember-3-22-released.md
Comment (ijlee2) [inline source/2020-10-19-ember-3-22-released.md]:
```suggestion
<!-- alex disable retext-equality -->
- Enabled Cache API, a low-level primitive for memoizing the result of a function based on autotracking. (You can install a [polyfill](https://github.com/ember-polyfills/ember-cache-primitive-polyfill) for apps between versions 3.13 and 3.21.) For more information, please check the [Ember Guides](https://guides.emberjs.com/release/in-depth-topics/autotracking-in-depth/#toc_caching-of-tracked-properties) and [Autotracking Memoization RFC](https://emberjs.github.io/rfcs/0615-autotracking-memoization.html).
```

File: source/2020-10-19-ember-3-22-released.md
Comment (MelSumner) [inline source/2020-10-19-ember-3-22-released.md]: This should be an ordered list to be consistent with the rest of the post.

**Review by MelSumner — APPROVED** [review]
Made one non-blocking suggestion, otherwise looks good. thank you for doing this work! 👍

File: source/2020-10-19-ember-3-22-released.md
Comment (ijlee2) [reply source/2020-10-19-ember-3-22-released.md]: Thanks! I will update the post soon.

**Comment by ijlee2** [comment]
@locks will merge this pull request after completing releasing API docs.

File: source/2020-10-19-ember-3-22-released.md
Comment (rwjblue) [inline source/2020-10-19-ember-3-22-released.md]: Not sure it matters to call out, but the 3.22 capability for modifier managers also allows modifiers to only re-run based on tracked state that they themselves consume. Previously, the framework would eagerly consume all of the arguments.

File: source/2020-10-19-ember-3-22-released.md
Comment (rwjblue) [inline source/2020-10-19-ember-3-22-released.md]: Is it possible to link to the published API docs we have for these? It would be pretty nice if we could...

File: source/2020-10-19-ember-3-22-released.md
Comment (rwjblue) [inline source/2020-10-19-ember-3-22-released.md]: I think I'd probably remove this bit if we didn't introduce any depreciations no need to tell them about the deprecation workflow...

Comment (ijlee2) [reply source/2020-10-19-ember-3-22-released.md]: Sounds good with me, thanks for your suggestion. Yeah, the paragraphs were a copy-paste from a template.

```suggestion
```

File: source/2020-10-19-ember-3-22-released.md
Comment (ijlee2) [inline source/2020-10-19-ember-3-22-released.md]:
```suggestion
- `RouterService#isActive` updates correctly when the route changes ([#19094](https://github.com/emberjs/ember.js/pull/19094))
- A custom modifier manager only consumes arguments that it actually uses. (Previously, the manager would eagerly consume all arguments.) In order for specific modifiers to take advantage of this change, they need to call `modifierCapabilities('3.22')` (instead of `'3.13'`). ([#19163](https://github.com/emberjs/ember.js/pull/19163))
- Modifier manager accepts the resolved value of `owner.factoryFor` directly. This approach aligns with how helpers and component managers work. ([#19170](https://github.com/emberjs/ember.js/pull/19170))
- Changes made to `protocolForUrl` (in `3.22.0-beta.2+`) work in `fastboot@3.1.0` environment by leveraging the native JavaScript `URL` global. ([#19124](https://github.com/emberjs/ember.js/pull/19124))
```

File: source/2020-10-19-ember-3-22-released.md
Comment (ijlee2) [reply source/2020-10-19-ember-3-22-released.md]: Thanks for pointing this out. I think we should mention it in the blog post.

File: source/2020-10-19-ember-3-22-released.md
Comment (ijlee2) [inline source/2020-10-19-ember-3-22-released.md]:
```suggestion
- `RouterService#isActive` updates correctly when the route changes. ([#19094](https://github.com/emberjs/ember.js/pull/19094))
```

**Comment by jenweber** [comment]
Thanks everyone for the work so far. I will see if I can get the docs up, and after they are published, this can be merged.

**Pull request merged**
